### PR TITLE
Experiment/changesets

### PR DESCRIPTION
**Pull request by ryan-roemer**
Part of #5 

## tasks

- [x] handle pnpm lock update on version / publish and integrate with scripts/docs

## This PR

- Basic changesets addition
- Document manual workflow
- I did a sample version and manual localdev publish of `v0.0.4`. You can see resultant changelogs produced (note we can change the format of this later with a lot of sophistication).

## Next up

- [ ] GH actions integration

### CODE REVIEW

**Review by gksander — APPROVED** [review]
Nice, looks awesome – looking forward to incorporating changesets! Just so I understand where we're going with this, we're looking to do all of our versioning/publishing via CI, correct? And that'll trigger via a merge to `main`? Or would there be a separate, dedicated release branch that gets merged to whenever we're ready to cut a release?

File: pnpm-wireit/README.md [hunk 1]
```diff
@@ -28,15 +28,67 @@ See some doggo info:
...
+
+_CI Note_: Our process is to batch up appropriate changes and then do these steps in a dedicated version branch / pull request.
+
+When all of the changes and changesets are ready for a new version, issue the following wrapper command:
+
+```sh
+$ pnpm run version
+```
+
+which should bump versions and write workspace CHANGELOG.md files ass appropriate. Review the git changes as appropriate, adjust anything amiss, and commit to git source.
```
[the remark this answers is not in the document]
Comment (ryan-roemer) [reply pnpm-wireit/README.md:65]: Oh my!

**Comment by ryan-roemer** [comment]
@gksander -- Yep, I think we'll do it like url:

1. Batch up changesets in PRs in main.
2. When we're ready to release, then do a version packages PR with `changeset version` that when merged does the actual `changeset publish`

File: pnpm-wireit/package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   },
   "homepage": "https://github.com/FormidableLabs/monorepo-infra-experiments#readme",
   "scripts": {
+    "version": "pnpm changeset version && pnpm install",
```
Comment (scottrippey) [inline pnpm-wireit/package.json:18]: What's `&& pnpm install` doing?

Comment (ryan-roemer) [reply pnpm-wireit/package.json:18]: Ah -- so the `changeset version` changes need an updated lockfile. I had CI fail for this reason and then googled and found https://github.com/changesets/changesets/issues/421

I do so wish that JSON allowed code comments 😛

Comment (ryan-roemer) [reply pnpm-wireit/package.json:18]: Actually, reading that more, I think I should be doing `pnpm install --lockfile-only`. I'll try that in next PR

**Pull request merged**